### ISSUE TITLE
--help doesn't work for python3.6

### ISSUE THREAD

**Issue by arthur-s** [pappasam/jedi-language-server#202]
with latest version:
```
 ~  $ jedi-language-server --help
Traceback (most recent call last):
  File "/home/arthur/.local/bin/jedi-language-server", line 5, in <module>
    from jedi_language_server.cli import cli
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi_language_server/cli.py", line 7, in <module>
    from .server import SERVER
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi_language_server/server.py", line 12, in <module>
    from jedi import Project
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/__init__.py", line 32, in <module>
    from jedi.api import Script, Interpreter, set_debug_function, preload_module
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/api/__init__.py", line 21, in <module>
    from jedi.api import classes
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/api/classes.py", line 24, in <module>
    from jedi.inference.utils import unite
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/inference/__init__.py", line 70, in <module>
    from jedi.inference import imports
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/inference/imports.py", line 23, in <module>
    from jedi.inference import compiled
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/inference/compiled/__init__.py", line 4, in <module>
    from jedi.inference.compiled.value import CompiledValue, CompiledName, \
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/inference/compiled/value.py", line 13, in <module>
    from jedi.inference.filters import AbstractFilter
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/inference/filters.py", line 17, in <module>
    from jedi.inference.names import TreeNameDefinition, ParamName, \
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/inference/names.py", line 11, in <module>
    from jedi.inference import docstrings
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/inference/docstrings.py", line 27, in <module>
    from jedi.inference.lazy_value import LazyKnownValues
  File "/home/arthur/.local/lib/python3.6/site-packages/jedi/inference/lazy_value.py", line 2, in <module>
    from jedi.common import monkeypatch
ImportError: cannot import name 'monkeypatch'
```
with python3.7 it works.

**Comment by pappasam** [comment]
@arthur-s thanks for raising this issue! Since Python 3.6 is [no longer officially supported](https://endoflife.date/python), and vscode appears to have also [stopped supporting Python 3.6](https://devblogs.microsoft.com/python/python-in-visual-studio-code-february-2022-release/#other-changes-and-enhancements), I figure we can address this by simply removing support for Python 3.6. Any strong objections?

**Comment by arthur-s** [comment]
@pappasam , no objections :)

**Comment by GitMensch** [comment]
Sadly the RedHat8 family, which has its EOL in Mid 2029 is fixed at Python 3.6.8 (and actually still provides security-updates for that).

The MS extension provides support for this again (if you want to debug, then you need to install an older version of only the sister debug extension).

Could the support for Python 3.6.8 be added?

I've only checked so far that the bundled importlib_metadata and typing_extensions are too new, dropping them and use a site-install [=older version] may work (for importlib a single import needs to by surrounded by try/catch).
